isis/src/clipper/apps/jitterfit/main.cpp

```diff
@@ -0,0 +1,234 @@
+#include "Isis.h"
+
+#include <iostream>
+
+#include <QList>
+#include <QString>
+
+#include "AutoReg.h"
+#include "AutoRegFactory.h"
+#include "BasisFunction.h"
+#include "Chip.h"
+#include "Cube.h"
+#include "CSVReader.h"
+#include "FileName.h"
+#include "LeastSquares.h"
+#include "NthOrderPolynomial.h"
+#include "Pvl.h"
+#include "Table.h"
+#include "UserInterface.h"
+
+using namespace std;
+using namespace Isis;
+
+struct RegistrationData {
+  int checkLine; //0
+  int checkSample; //1
+  double checkTime; //2
+  double matchedLine; //3
+  double matchedSample; //4
+  double matchedTime; //5
+  double deltaLine; //6
+  double deltaSample; //7
+  double goodness; //8 Goodness of fit
+  int success; //9
+};
+
+void IsisMain() {
+  
+  bool registrationFileSpecified = false;
+  
+  UserInterface &ui = Application::GetUserInterface();
+  
+  Cube jitterCube;
+  jitterCube.open(ui.GetFileName("FROM"), "rw");
+  
+  Cube checkCube;
+  checkCube.open(ui.GetFileName("FROM2"), "r");
+  
+  Pvl defFile;
+  defFile.read(ui.GetFileName("DEFFILE"));
+  AutoReg *ar = AutoRegFactory::Create(defFile);
+
+  double scale = ui.GetDouble("SCALE");
+
+  int pointSpacing = jitterCube.sampleCount();
+
+  // Setup the registration results file
+  ofstream outputFile;
+  if (ui.WasEntered("TO")) {
+    registrationFileSpecified = true;
+    QString to(FileName(ui.GetFileName("TO")).expanded());
+    outputFile.open(to.toLatin1().data());
+    outputFile << "# checkline line, checkline sample, checkline time taken, "
+                  "matched jittered image line, matched jittered image "
+                  "sample, matched jittered image time taken, delta line, "
+                  "delta sample, goodness of fit, registration success " << endl;
+  }
+
+  // ???? Question: Why use a file name here? Can't Table/blob read from an open Cube?
+  Table mainReadouts(QString("Normalized Main Readout Line Times"), jitterCube.fileName());
+  Table checklineReadouts(QString("Normalized Checkline Readout Line Times"), checkCube.fileName());
```

Comment:
Did this question ever get resolved?

Reply:
Not yet?

Reply:
Tables cannot read from Cubes, but Cubes can read their Tables.

https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/base/objs/Cube/Cube.h#L250

I am concerned about the conversion to a reference here, though. I think it still works, but I'm not 100% sure.

Reply:
That's probably what I was remembering. Improving this didn't make the deadline. I'll put fixing this on the list of tasks for next time.